File: exemples/typescript/123sms.ts
// 123sms.ts — Envoi de SMS professionnel via l'API 123-SMS.net
// Copyright (C) 123-Sms.net — licence MIT
// Node.js 18+ (fetch natif), Deno ou Bun. Aucune dependance npm.
//
// Execution :  npx tsx 123sms.ts     (ou : deno run --allow-net 123sms.ts)
// Identifiants : espace client https://www.123-sms.net > rubrique API.

const EMAIL = "votre-email@exemple.fr";
const CLE_API = "CLEAPI";
const API_URL = "https://www.123-sms.net/http.php";

/** Codes retour documentes de l'API 123-SMS. */
type CodeRetour = "80" | "81" | "82" | "83" | "84" | string;

const LIBELLES: Record<string, string> = {
  "80": "le message a ete envoye.",
  "81": "enregistre pour un envoi differe.",
  "82": "identifiants invalides.",
  "83": "credit insuffisant.",
  "84": "numero invalide.",
};

/** Envoie un SMS et renvoie le code retour de l'API (80 = envoye). */
export async function envoyerSms(
  numero: string,
  message: string,
  options?: { sender?: string; refaccuse?: string },
): Promise<CodeRetour> {
  const corps = new URLSearchParams({
    email: EMAIL,
    pass: CLE_API,
    numero,
    message,
    ...(options?.sender ? { sender: options.sender } : {}),
    ...(options?.refaccuse ? { refaccuse: options.refaccuse } : {}),
  });
  // POST en HTTPS : la cle API ne transite jamais en clair
  const reponse = await fetch(API_URL, {
    method: "POST",
    headers: { "Content-Type": "application/x-www-form-urlencoded" },
    body: corps.toString(),
    signal: AbortSignal.timeout(15_000),
  });
  return (await reponse.text()).trim();
}

// Demonstration
const code = await envoyerSms("33601020304", "Bonjour, ceci est un test 123-SMS.");
console.log(`Code ${code} : ${LIBELLES[code] ?? "voir la documentation."}`);

File: exemples/csharp/123sms.cs
// 123sms.cs — Envoi de SMS professionnel via l'API 123-SMS.net
// Copyright (C) 123-Sms.net — licence MIT
// Compatible .NET 6/7/8 et .NET Framework 4.6+ (HttpClient).
//
// Compilation rapide (.NET SDK) :  dotnet run
// ou creer un projet console et coller ce fichier.
//
// Parametres (espace client https://www.123-sms.net > rubrique API) :
//   email   : e-mail du compte 123-SMS
//   pass    : cle API (6 caracteres)
//   numero  : destinataire(s), ex. "33601020304" (plusieurs : separes par -)
//   message : texte du SMS (160 caracteres GSM par SMS)
// Optionnels : sender (Sender-ID declare), refaccuse (accuse par HTTP)

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Threading.Tasks;

class Sms123
{
    static async Task Main()
    {
        var parametres = new Dictionary<string, string>
        {
            ["email"]   = "votre-email@exemple.fr",
            ["pass"]    = "CLEAPI",
            ["numero"]  = "33601020304",
            ["message"] = "Bonjour, ceci est un test 123-SMS."
            // ["sender"]    = "MASOCIETE",
            // ["refaccuse"] = "commande-1234",
        };

        using var client = new HttpClient { Timeout = TimeSpan.FromSeconds(15) };
        using var contenu = new FormUrlEncodedContent(parametres);

        // POST en HTTPS : les identifiants ne transitent jamais en clair
        var reponse = await client.PostAsync("https://www.123-sms.net/http.php", contenu);
        string code = (await reponse.Content.ReadAsStringAsync()).Trim();

        Console.WriteLine(code switch
        {
            "80" => "80 : le message a ete envoye.",
            "81" => "81 : enregistre pour un envoi en differe.",
            "82" => "82 : identifiants invalides.",
            "83" => "83 : credit insuffisant.",
            "84" => "84 : numero invalide.",
            _    => "Code retour API : " + code
        });
    }
}
